pkg/actions/db_service.go

```diff
@@ -402,3 +414,134 @@ func (s *DBService) NewRunID() string {
 	tm := time.Now().UTC().Format(graveler.RunIDTimeLayout)
 	return tm + id
 }
+
+type runResultMigrate struct {
+	RunResult
+	RepoID string `db:"repository_id" json:"repository_id"`
+}
+
+type taskResultMigrate struct {
+	TaskResult
+	RepoID string `db:"repository_id" json:"repository_id"`
+}
+
+func Migrate(ctx context.Context, d *pgxpool.Pool, writer io.Writer) error {
+	je := json.NewEncoder(writer)
+	// Create header
+	err := je.Encode(kv.Header{
+		LakeFSVersion:   version.Version,
+		PackageName:     packageName,
+		DBSchemaVersion: kv.InitialMigrateVersion,
+		CreatedAt:       time.Now().UTC(),
+	})
+	if err != nil {
+		return err
+	}
+
+	// Dump RunResults
+	rows, err := d.Query(ctx, "SELECT * FROM actions_runs ORDER BY run_id DESC")
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+	rowScanner := pgxscan.NewRowScanner(rows)
+	for rows.Next() {
+		newRunID := graveler.NewRunID()
+		m := runResultMigrate{}
+		err = rowScanner.Scan(&m)
+		if err != nil {
+			return err
+		}
+
+		oldRunID := m.RunResult.RunID
+		m.RunResult.RunID = newRunID
+		pr := protoFromRunResult(&m.RunResult)
+		value, err := proto.Marshal(pr)
+		if err != nil {
+			return err
+		}
+		key := []byte(RunPath(m.RepoID, newRunID))
+		err = je.Encode(kv.Entry{
+			PartitionKey: []byte(PartitionKey),
+			Key:          key,
+			Value:        value,
+		})
+		if err != nil {
+			return err
+		}
+
+		// Add secondary keys branch
+		if m.BranchID != "" {
+			secKey := []byte(RunByBranchPath(m.RepoID, m.BranchID, newRunID))
```

Comment:
I think we should leverage the code that exists in the Actions KV implementation to handle all of this, instead of copying it.

Reply:
I'm struggling with trying to reuse the code in the service - do you have any suggestions?